pages/nfttest/fetch/index.vue

```diff
@@ -102,11 +106,52 @@ export default class FetchIndex extends Vue {
   url = this.$route.query.url as string || ''
   ownerWallet = ''
   errorMessage = ''
-  state = 'init'
+  crawledData: any
+  txHash = ''
+  ipfsHash = ''
+  arweaveId = ''
+  arweaveFeeTargetAddress = ''
+  arweaveFee = new BigNumber(0)
+  memo = ''
   iscnId = this.$route.query.iscn_id as string || ''
   isLoading = false
   avatar = null;
 
+  get formData(): FormData | null {
+    if (!this.crawledData?.body) { return null }
+    const body = new Blob([this.crawledData.body], { type: "text/html" })
+    const formData = new FormData()
+    formData.append('file', body, 'index.html')
+    return formData
+  }
+
+  get iscnPayload(): ISCNRegisterPayload {
+    const { title, keywords } = this.crawledData
+    let { description } = this.crawledData
+    description = this.truncate(description, 200)
+    return {
+      type: 'CreativeWork',
+      name: title,
+      description,
+      tagsString: keywords,
+      url: this.url,
+      exifInfo: {},
+      license: '',
+      ipfsHash: this.ipfsHash,
+      arweaveId: this.arweaveId,
```

Comment:
Do we prefer `undefined` here if they are not set?

Reply:
hmmm I think using `if` to guard it should be more clear

Reply:
return undefined would work if they are computed